packages/components/src/components/SquiggleChart.tsx

```diff
@@ -191,30 +204,24 @@ export interface SquiggleChartProps {
   /** If the output requires monte carlo sampling, the amount of samples */
   sampleCount?: number;
   /** The amount of points returned to draw the distribution */
-  outputXYPoints?: number;
-  kernelWidth?: number;
-  pointDistLength?: number;
-  /** If the result is a function, where the function starts */
-  diagramStart?: number;
-  /** If the result is a function, where the function ends */
-  diagramStop?: number;
-  /** If the result is a function, how many points along the function it samples */
-  diagramCount?: number;
+  environment: environment;
+  /** If the result is a function, where the function starts, ends and the amount of stops */
+  chartSettings?: FunctionChartSettings;
   /** When the environment changes */
   onChange?(expr: squiggleExpression): void;
   /** CSS width of the element */
   width?: number;
   height?: number;
   /** Bindings of previous variables declared */
-  bindings?: bindings;
+  bindings: bindings;
   /** JS imported parameters */
   jsImports?: jsImports;
   /** Whether to show a summary of the distirbution */
   showSummary?: boolean;
   /** Whether to show type information about returns, default false */
-  showTypes?: boolean;
+  showTypes: boolean;
   /** Whether to show graph controls (scale etc)*/
-  showControls?: boolean;
+  showControls: boolean;
```

Comment:
what's the reasoning behind forcing the user to provide/commit instead of default to false?